src/api/models/ListingItem.ts

```diff
@@ -93,19 +93,28 @@ export class ListingItem extends Bookshelf.Model<ListingItem> {
         }
     }
 
-    public static async searchByCategoryOrName(categoryId: number, searchTerm: string = '', withRelated: boolean = true): Promise<Collection<ListingItem>> {
+    public static async searchBy(options: any): Promise<Collection<ListingItem>> {
```

Comment:
add the withRelated as the second parameter, that way the options contains only search options that can be specified by the user.

also specify the class ListingItemSearchParams instead of any.